### PR TITLE
Add legacy code post

### LINKED ISSUE #3: Create working with legacy code blog post
Write a blog post on "working with legacy code".

### PR DESCRIPTION
**Pull request by jonmaddock**
Closes #3.

### CODE REVIEW

**Comment by jonmaddock** [comment]
The bulk of the post has been added, but we still need to add the case studies and further reading/links.

File: _posts/2021-04-13-working-with-legacy-code.md [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+layout: post
+title: "Working with legacy code"
+date: 2021-04-13 12:00:00 +0000
+categories: legacy
+---
+## What is legacy code?
+There is some debate about what exactly constitutes “legacy code”, but it is generally older code which is difficult to maintain. We expand below on what makes code difficult to maintain and suggest a general approach to improving it.
+
+## What makes code difficult to maintain? 
+There are many articles written about software best practices and how to make code easier to maintain.  Example features of legacy code include:
```
Comment (johnnonweiler) [inline _posts/2021-04-13-working-with-legacy-code.md]: I suggest deleting (or moving) "There are many articles written about software best practices and how to make code easier to maintain."  - It doesn't add much, and with the previous paragraph starting with "there is some debate about...", it reads a little like we're trying to carefully qualify everything.

**Review by johnnonweiler — COMMENTED** [review]
Looks good to me!  (And we should talk about publishing it.)

File: _posts/2021-04-13-working-with-legacy-code.md (hunk 1 above)
Comment (johnnonweiler) [reply _posts/2021-04-13-working-with-legacy-code.md]: (We were talking about adding some links to other articles, which might be a better way of putting this.)

Comment (jonmaddock) [reply _posts/2021-04-13-working-with-legacy-code.md]: I've deleted the above line about "there are many articles..."

**Comment by jonmaddock** [comment]
I've also linked to Jonathan H's UDA post. Are we happy to merge in time for my talk tomorrow?

**Pull request merged**